packages/grafana-data/src/field/fieldColor.ts

```diff
@@ -0,0 +1,87 @@
+import { FALLBACK_COLOR, Field, FieldColorModeId, GrafanaTheme, Threshold } from '../types';
+import { classicColors, getColorFromHexRgbOrName, Registry, RegistryItem } from '../utils';
+import { interpolateRgbBasis } from 'd3-interpolate';
+import { fallBackTreshold } from './thresholds';
+
+export type FieldValueColorCalculator = (value: number, percent: number, Threshold?: Threshold) => string;
+
+export interface FieldColorMode extends RegistryItem {
+  getCalculator: (field: Field, seriesIndex: number, theme: GrafanaTheme) => FieldValueColorCalculator;
+}
+
+export const fieldColorModeRegistry = new Registry<FieldColorMode>(() => {
+  return [
+    {
+      id: FieldColorModeId.Thresholds,
+      name: 'Discrete / From thresholds',
```

Comment:
Maybe just "From thresholds", if we add a continuous mode, it could be:

"From thresholds (continuous)"